src/file_management/core.clj

```diff
@@ -73,11 +75,15 @@
            - file-value {:value ;file content
                          :content-type }
    Returns: file url to s3"
-  (let [file-key (gets-file-key file-key)]
-    (s3/put-object credentials
-                   (:bucket credentials)
-                   file-key
-                   (:value file)
-                   {:content-type (:content-type file)}
-                   (s3/grant :all-users :read))
-    (get-file-url credentials file-key))))
+  (go
+    (let [file-key (gets-file-key file-key)]
+      (try
+        (s3/put-object credentials
+                       (:bucket credentials)
+                       file-key
+                       (:value file)
+                       {:content-type (:content-type file)}
+                       (s3/grant :all-users :read))
+        (result/success (get-file-url credentials file-key))
```

Comment:
if put-object it throws an exception, it could be AmazonClientException and AmazonServiceException.

Ref: http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#putObject(com.amazonaws.services.s3.model.PutObjectRequest)